app/models/user_access.rb

```diff
@@ -59,6 +59,63 @@ def can?(action, model, record = nil)
     end
   end
 
+  def access_tree(action)
```

Comment:
This method looks like it's only for views (and it probably is), but it's better documented and tested and placed here because it's a critical piece that makes important DB queries and must be well tested (like any other `UserAccess` method).